shell/platform/linux/fl_string_codec_test.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_string_codec.h"
+#include "flutter/shell/platform/linux/fl_test.h"
+#include "gtest/gtest.h"
+
+static gchar* encode_value(FlValue* value) {
+  g_autoptr(FlStringCodec) codec = fl_string_codec_new();
+  g_autoptr(GError) error = nullptr;
+  g_autoptr(GBytes) message =
+      fl_message_codec_encode_message(FL_MESSAGE_CODEC(codec), value, &error);
+  EXPECT_NE(message, nullptr);
+  EXPECT_EQ(error, nullptr);
+
+  return bytes_to_hex_string(message);
+}
+
+static void encode_value_error(FlValue* value, GQuark domain, int code) {
+  g_autoptr(FlStringCodec) codec = fl_string_codec_new();
+  g_autoptr(GError) error = nullptr;
+  g_autoptr(GBytes) message =
+      fl_message_codec_encode_message(FL_MESSAGE_CODEC(codec), value, &error);
+  EXPECT_EQ(message, nullptr);
+  EXPECT_TRUE(g_error_matches(error, domain, code));
+}
+
+static FlValue* decode_value(const char* hex_string) {
+  g_autoptr(FlStringCodec) codec = fl_string_codec_new();
+  g_autoptr(GBytes) message = hex_string_to_bytes(hex_string);
+  g_autoptr(GError) error = nullptr;
+  g_autoptr(FlValue) value =
+      fl_message_codec_decode_message(FL_MESSAGE_CODEC(codec), message, &error);
+  EXPECT_EQ(error, nullptr);
+  EXPECT_NE(value, nullptr);
+  return fl_value_ref(value);
+}
+
+TEST(FlStringCodecTest, EncodeData) {
+  g_autoptr(FlValue) value = fl_value_string_new("hello");
+  g_autofree gchar* hex_string = encode_value(value);
+  EXPECT_STREQ(hex_string, "68656c6c6f");
```

Comment:
Out of curiosity, why the hex strings rather than just a raw byte array? It's shorter, but doesn't seem more readable (vs [this](https://github.com/flutter/engine/blob/9992a4ae664b0de4bf1834ad44dc48c7ee6025e8/shell/platform/common/cpp/client_wrapper/standard_message_codec_unittests.cc#L84))

Reply:
It just seemed harder to do it directly with `GBytes`:
```c
g_autoptr(FlValue) value = fl_value_string_new("hello");
g_autoptr(GBytes) message = encode_value(value);
uint8_t data[] = {0x64, 0x65, 0x6c, 0x6c, 0x6f};
g_autoptr(GBytes) expected = g_bytes_new (data, 5);
EXPECT_TRUE(g_bytes_equal(message, value));
```

There's the risk of stuffing up the length and the test case won't print out something easily readable if they don't match.

If there's a cleverer way to do it with gtest happy to do that, only did it this way for practical reasons.